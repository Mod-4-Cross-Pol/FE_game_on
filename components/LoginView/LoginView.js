import React from 'react';
import { StyleSheet, Text, View, Button, TextInput, Image } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

const Stack = createStackNavigator();

export default function LoginView({ navigation }) {
  return (
    <View style={styles.container}>
       <Image
         style={styles.logo}
         source={require('../../assets/game_on_logo.png')}
       />
      <View style={styles.viewInput}><TextInput style={styles.input} placeholder='username...'/></View>
      <View style={styles.viewInput}><TextInput style={styles.input} placeholder='password...' /></View>
      <View style={styles.signInButton}><Button onPress={() => navigation.navigate('Dashboard')} title='SIGN IN'/></View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#292c33',
    alignItems: 'center'
  },
  logo: {
    height: 300,
    width: '100%',
    resizeMode: 'contain',
    marginTop: '30%',
    marginLeft: 15
  },
  input: {
    height: 50,
    width: 300,
    backgroundColor: '#fff',
    borderRadius: 25,
    textAlign: 'center',
    fontSize: 20
  },
  viewInput: {
    marginTop: 20
  },
  signInButton: {
    marginTop: 20,
    width: 200
  },

});
