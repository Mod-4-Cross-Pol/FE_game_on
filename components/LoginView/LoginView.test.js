import React from 'react';
import { LoginView } from './LoginView';
import renderer from 'react-test-renderer';

describe('LoginView', () => {

  describe('Renders a LoginView component', () => {

    test('renders correctly', () => {
      const wrapper = renderer.create(<LoginView />).toJSON();
      expect(wrapper).toMatchSnapshot();
    });

  });

});
