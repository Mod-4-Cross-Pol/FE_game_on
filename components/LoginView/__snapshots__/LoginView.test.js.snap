// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView Renders a LoginView component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#292c33",
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 300,
        "marginLeft": 15,
        "marginTop": "30%",
        "resizeMode": "contain",
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="username..."
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 25,
          "fontSize": 20,
          "height": 50,
          "textAlign": "center",
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="password..."
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 25,
          "fontSize": 20,
          "height": 50,
          "textAlign": "center",
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": 200,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          SIGN IN
        </Text>
      </View>
    </View>
  </View>
</View>
`;
